src/combat.ts

```diff
@@ -130,52 +145,262 @@ export class Macro extends LibramMacro {
+          Macro.while_("!pastround 3 && !hppercentbelow 25", Macro.item("seal tooth"))
+        )
+        .externalIf(
+          myFamiliar() === $familiar`Stocking Mimic`,
+          Macro.while_("!pastround 10 && !hppercentbelow 25", Macro.item("seal tooth"))
+        )
+        .externalIf(
+          myFamiliar() === $familiar`Hobo Monkey`,
+          Macro.while_(
+            `!match "shoulder, and hands you some Meat." && !pastround 20 && !hppercentbelow 25`,
+            Macro.item("seal tooth")
+          )
+        )
+        .tryItem($item`porquoise-handled sixgun`);
+    }
+
+    // Only stasis if the monster manuel is available and we have access to monsterhpabove
+    if (!monsterManuelAvailable()) {
+      return this;
+    }
+    const passiveDamage = maxPassiveDamage() + 5;
+
+    // Ignore unexpected monsters, holiday scaling monsters seem to abort with monsterhpabove
+    return this.if_(
+      "monstername angry tourist || monstername garbage tourist || monstername horrible tourist family || monstername Knob Goblin Embezzler || monstername sausage goblin",
+      Macro.if_(`monsterhpabove ${passiveDamage}`, Macro.trySkill("Pocket Crumbs"))
+        .if_(`monsterhpabove ${passiveDamage}`, Macro.trySkill("Extract"))
+        .externalIf(
+          haveEquipped($item`Buddy Bjorn`) || haveEquipped($item`Crown of Thrones`),
+          Macro.while_(`!pastround 3 && monsterhpabove ${passiveDamage}`, Macro.item("seal tooth"))
+        )
+        .externalIf(
+          myFamiliar() === $familiar`Stocking Mimic`,
+          Macro.while_(`!pastround 10 && monsterhpabove ${passiveDamage}`, Macro.item("seal tooth"))
+        )
+        .externalIf(
+          myFamiliar() === $familiar`Hobo Monkey`,
+          Macro.while_(
+            `!match "shoulder, and hands you some Meat." && !pastround 20 && monsterhpabove ${passiveDamage}`,
+            Macro.item("seal tooth")
+          )
+        )
+        .if_(
+          `monsterhpabove ${passiveDamage + 40}`,
+          Macro.tryHaveItem($item`porquoise-handled sixgun`)
+        )
+    );
+  }
+
+  static meatStasis(checkPassive: boolean): Macro {
+    return new Macro().meatStasis(checkPassive);
+  }
+
+  startCombat(): Macro {
+    return this.tryHaveSkill($skill`Sing Along`)
+      .tryHaveSkill($skill`Curse of Weaksauce`)
+      .trySkill($skill`Pocket Crumbs`)
+      .trySkill($skill`Extract`)
+      .tryHaveItem($item`porquoise-handled sixgun`)
+      .externalIf(have($skill`Meteor Lore`), Macro.trySkill($skill`Micrometeorite`))
+      .tryHaveItem($item`Time-Spinner`)
+      .tryHaveItem($item`Rain-Doh indigo cup`)
+      .tryHaveItem($item`Rain-Doh blue balls`)
+      .externalIf(
+        haveEquipped($item`Buddy Bjorn`) || haveEquipped($item`Crown of Thrones`),
+        Macro.while_("!pastround 3 && !hppercentbelow 25", Macro.item("seal tooth"))
+      )
+      .externalIf(
+        myFamiliar() === $familiar`Stocking Mimic`,
```

Comment:
Could we add NPZR to this?

Reply:
yeah, how many rounds should it stasis for